src/main/java/com/gotocompany/depot/maxcompute/converter/type/StructTypeInfoConverter.java

```diff
@@ -0,0 +1,22 @@
+package com.gotocompany.depot.maxcompute.converter.type;
+
+import com.aliyun.odps.type.TypeInfo;
+import com.aliyun.odps.type.TypeInfoFactory;
+import com.google.protobuf.Descriptors;
+
+public class StructTypeInfoConverter implements TypeInfoConverter {
+
+    private static final String GOOGLE_PROTOBUF_STRUCT = "google.protobuf.Struct";
+
+    @Override
+    public TypeInfo convertSingular(Descriptors.FieldDescriptor fieldDescriptor) {
+        return TypeInfoFactory.STRING;
+    }
+
+    @Override
+    public boolean canConvert(Descriptors.FieldDescriptor fieldDescriptor) {
+        return fieldDescriptor.getType() == Descriptors.FieldDescriptor.Type.MESSAGE
+                && fieldDescriptor.getMessageType().getFullName().equals(GOOGLE_PROTOBUF_STRUCT);
```

Comment:
No handling for empty structs or structs with null fields. Also missing validation for maximum struct size and nested struct depth. Should add proper null checks and size limits.

Reply:
need to check the consistency with BQ one for empty message

Reply:
empty struct will be mapped -> {}
null fields will be null